cloud/azure/management/costs/mode/costsexplorer.pm

```diff
@@ -0,0 +1,290 @@
+# limitations under the License.
+#
+
+package cloud::azure::management::costs::mode::costsexplorer;
+
+use base qw(centreon::plugins::templates::counter);
+
+use strict;
+use warnings;
+use DateTime;
+
+sub resource_group_prefix_output {
+    my ($self, %options) = @_;
+
+    return sprintf( "Resource group '%s' ", $options{instance});
+}
+
+sub custom_subscription_output {
+    my ($self, %options) = @_;
+
+    return sprintf( "Subscription costs for specified period: %.2f %s", $self->{result_values}->{subscription_cost}, $self->{result_values}->{currency});
+}
+
+sub custom_resource_group_output {
+    my ($self, %options) = @_;
+
+    return sprintf( "costs for specified period: %.2f %s", $self->{result_values}->{resource_group_cost}, $self->{result_values}->{currency});
+}
+
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 0, skipped_code => { -10 => 1 } },
+        { name => 'resource_group', type => 1, cb_prefix_output => 'resource_group_prefix_output', message_multiple => 'All resource group costs are OK', skipped_code => { -10 => 1 } },
+    ];
+
+    $self->{maps_counters}->{global} = [
+        { label => 'subscription-costs', nlabel => 'subscription.global.costs', set => {
+                key_values => [ { name => 'subscription_cost' }, { name => 'currency' } ],
+                closure_custom_output => $self->can('custom_subscription_output'),
+                perfdatas => [
+                    { template => '%.2f', min => 0 }
+                ]
+            }
+        }
+    ];
+    $self->{maps_counters}->{resource_group} = [
+        { label => 'resource-group-costs', nlabel => 'resourcegroup.costs', set => {
+                key_values => [ { name => 'resource_group_cost' }, { name => 'currency'} ],
+                closure_custom_output => $self->can('custom_resource_group_output'),
+                perfdatas => [
+                    { template => '%.2f', min => 0, label_extra_instance => 1 },
+                ]
+            }
+        }
+    ]
+
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        "lookup-days:s"          => { name => 'lookup_days', default => 30 },
+        "resource-group:s@"      => { name => 'resource_group' },
+        "tags:s@"                => { name => 'tags' }
+    });
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+
+    if (defined($self->{option_results}->{tags}) && $self->{option_results}->{tags} ne '') {
+        foreach my $tag_pair (@{$self->{option_results}->{tags}}) {
+            my ($key, $value) = split / => /, $tag_pair;
+            $value = "" if !defined($value);
+            centreon::plugins::misc::trim($value) if defined($value);
```

Comment:
Not sure when $value can be undefined then. Line 101 condition sounds weird but I'm probably missing stg

Reply:
Right. It cannot be undefined